src/collection-view.js

```diff
@@ -321,9 +337,19 @@ Marionette.CollectionView = Marionette.View.extend({
     // remove and/or close it later
     this.children.add(view);
 
+    if (canTriggerAttach && this.triggerBeforeAttach) {
+      nestedViews = [view].concat(view._getNestedViews());
+      // Trigger before:attach after render to honor Region#show() lifecycle order.
```

Comment:
Good question!  I didn't think about the default return value for _getNestedView().  It appears [we can rely on an empty array in the empty case, however](https://github.com/ianmstew/backbone.marionette/blob/9907c2982fa57138d1a29596cd8dc41925ff32b2/src/view.js#L321-L337).